src/rgw/rgw_rest_s3.cc

```diff
@@ -661,50 +661,74 @@ void RGWGetUsage_ObjStore_S3::send_response()
   rgw_flush_formatter_and_reset(s, s->formatter);
 }
 
-int RGWListBucket_ObjStore_S3::get_params()
+
+
+int RGWListBucket_ObjStore_S3::get_common_params()
+
 {
-  list_versions = s->info.args.exists("versions");
-  prefix = s->info.args.get("prefix");
-  if (!list_versions) {
-    marker = s->info.args.get("marker");
-  } else {
-    marker.name = s->info.args.get("key-marker");
-    marker.instance = s->info.args.get("version-id-marker");
-  }
+list_versions = s->info.args.exists("versions");
+prefix = s->info.args.get("prefix");
 
-  // non-standard
-  s->info.args.get_bool("allow-unordered", &allow_unordered, false);
+// non-standard
+s->info.args.get_bool("allow-unordered", &allow_unordered, false);
+delimiter = s->info.args.get("delimiter");
+max_keys = s->info.args.get("max-keys");
+op_ret = parse_max_keys();
+if (op_ret < 0) {
+return op_ret;
+}
 
-  delimiter = s->info.args.get("delimiter");
+encoding_type = s->info.args.get("encoding-type");
+if (s->system_request) {
+s->info.args.get_bool("objs-container", &objs_container, false);
+const char *shard_id_str = s->info.env->get("HTTP_RGWX_SHARD_ID");
+if (shard_id_str) {
+string err;
+shard_id = strict_strtol(shard_id_str, 10, &err);
+if (!err.empty()) {
+ldout(s->cct, 5) << "bad shard id specified: " << shard_id_str << dendl;
+return -EINVAL;
+}
+} 
+else
+{
+shard_id = s->bucket_instance_shard_id;
+}
+}
+return 0;
+}
 
-  max_keys = s->info.args.get("max-keys");
-  op_ret = parse_max_keys();
-  if (op_ret < 0) {
-    return op_ret;
-  }
+int RGWListBucket_ObjStore_S3::get_params()
+{
+int ret = get_common_params();
+if (ret < 0)
+return ret;
+if (!list_versions)
+{
+marker = s->info.args.get("marker");
+} else {
+marker.name = s->info.args.get("key-marker");
+marker.instance = s->info.args.get("version-id-marker");
+}
+return 0;
+}
 
-  encoding_type = s->info.args.get("encoding-type");
-  if (s->system_request) {
-    s->info.args.get_bool("objs-container", &objs_container, false);
-    const char *shard_id_str = s->info.env->get("HTTP_RGWX_SHARD_ID");
-    if (shard_id_str) {
-      string err;
-      shard_id = strict_strtol(shard_id_str, 10, &err);
-      if (!err.empty()) {
-        ldout(s->cct, 5) << "bad shard id specified: " << shard_id_str << dendl;
-        return -EINVAL;
-      }
-    } else {
-      shard_id = s->bucket_instance_shard_id;
-    }
-  }
 
-  return 0;
+int RGWListBucket_ObjStore_S3v2::get_params()
+{
+int ret = get_common_params();
+if (ret < 0)
+return ret;
+s->info.args.get_bool("fetchOwner", &fetchOwner, false);
```

Comment:
according to [amazon docs](https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html#v2-RESTBucketGET-requests-request-parameters), this parameter is called `fetch-owner`

Reply:
Thanks. Modified.